android/Staccato_AN/app/src/main/java/com/on/staccato/presentation/bindingadapter/TextBindingAdapter.kt

```diff
@@ -0,0 +1,158 @@
+        setTextColor(resources.getColor(R.color.gray3, null))
+        isClickable = false
+        isFocusable = false
+    } else if (selectedMemory == null) {
+        text = resources.getString(R.string.visit_creation_memory_selection_hint)
+        setTextColor(resources.getColor(R.color.gray3, null))
+        isClickable = true
+        isFocusable = true
+    } else {
+        text = selectedMemory.memoryTitle
+        setTextColor(resources.getColor(R.color.staccato_black, null))
+    }
+}
+
+@BindingAdapter(value = ["dateTimeWithAmPm", "memoryCandidates"])
+fun TextView.setDateTimeWithAmPm(
+    dateTime: LocalDateTime?,
+    memoryCandidates: MemoryCandidates?,
+) {
+    if (memoryCandidates?.memoryCandidate?.isEmpty() == true) {
+        text = resources.getString(R.string.visit_creation_memory_selection_hint)
+        setTextColor(resources.getColor(R.color.gray3, null))
+        isClickable = false
+        isFocusable = false
+    } else {
+        text = dateTime?.let(::getFormattedLocalDateTime) ?: ""
+        setTextColor(resources.getColor(R.color.staccato_black, null))
+        isClickable = true
+        isFocusable = true
+    }
+}
+
+@BindingAdapter(
+    value = ["startDate", "endDate"],
+)
+fun TextView.setMemoryPeriod(
+    startDate: LocalDate?,
+    endDate: LocalDate?,
+) {
+    if (startDate == null || endDate == null) {
+        text = resources.getString(R.string.memory_creation_period_hint)
+        setTextColor(resources.getColor(R.color.gray3, null))
+    } else {
+        text =
+            resources.getString(R.string.memory_creation_selected_period)
+                .format(startDate, endDate)
+        setTextColor(resources.getColor(R.color.staccato_black, null))
+    }
+}
+
+@BindingAdapter("isMemoryCandidatesEmpty")
+fun TextView.setIsMemoryCandidatesEmptyVisibility(memoryCandidates: MemoryCandidates?) {
+    isGone = !memoryCandidates?.memoryCandidate.isNullOrEmpty()
+}
+
+@BindingAdapter("isMemoryEmpty")
+fun TextView.setIsMemoryEmptyVisibility(items: List<Int>?) {
+    isGone = !items.isNullOrEmpty()
+}
+
+@BindingAdapter("visitedAtHistory")
+fun TextView.formatVisitedAtHistory(visitedAt: LocalDateTime?) {
+    text = visitedAt?.let {
+        getFormattedLocalDateTime(it) + resources.getString(R.string.visit_at_history)
+    } ?: ""
+}
+
+@BindingAdapter(
+    value = ["startAt", "endAt"],
+)
+fun TextView.formatLocalDateToDatePeriod(
+    startAt: LocalDate?,
+    endAt: LocalDate?,
+) {
+    val periodFormatString = resources.getString(R.string.memory_period_dot)
+    text =
+        if (startAt != null && endAt != null) {
+            visibility = View.VISIBLE
+            format(
+                periodFormatString,
+                startAt.year,
+                startAt.monthValue,
+                startAt.dayOfMonth,
+                endAt.year,
+                endAt.monthValue,
+                endAt.dayOfMonth,
+            )
```

Comment:
해당 바인딩 어댑터에서 사용하는 format 함수가 `okhttp3` 라이브러리의 `Util.kt` 에 존재하는 포맷팅 메서드네요...!
전역에서 편하게 사용하기 위해 만들어진 Util 이지만, `okhttp3` 와 관련이 없는 `BindingAdatpers` 에 해당 라이브러리의 import가 존재하는 것이 불필요한 의존성을 만든 것 같아 좋은 구현이라고 생각하지 않습니다.
`String`의 확장함수로 정의되어있는 `String.format()` 메서드 등 다른 활용 방안이 많으니, 이로 대체하는 것이 좋아보입니다!

```suggestion
            periodFormatString.format(
                startAt.year,
                startAt.monthValue,
                startAt.dayOfMonth,
                endAt.year,
                endAt.monthValue,
                endAt.dayOfMonth,
            )
```

Reply:
띠용~ okhttp3의 format import가 숨어있었네요!!
제안해주신대로 String.format()로 변경했습니다 👍